test/e2e/tracing_test.go

```diff
@@ -0,0 +1,119 @@
+	env, err := e2e.NewDockerEnvironment("e2e-tracing-test")
+	testutil.Ok(t, err)
+	t.Cleanup(env.Close)
+	name := "testing"
+	newJaeger := e2e.NewInstrumentedRunnable(env, fmt.Sprintf("jaeger-%s", name)).
+		WithPorts(
+			map[string]int{
+				"http":                      16686,
+				"http.admin":                14269,
+				"jaeger.thrift-model.proto": 14250,
+				"jaeger.thrift":             14268,
+			}, "http.admin").
+		Init(e2e.StartOptions{
+			Image:     "jaegertracing/all-in-one:1.33",
+			Readiness: e2e.NewHTTPReadinessProbe("http.admin", "/", 200, 200),
+		})
+	testutil.Ok(t, e2e.StartAndWaitReady(newJaeger))
+
+	jaegerConfig, err := yaml.Marshal(client.TracingConfig{
+		Type: client.Jaeger,
+		Config: jaeger.Config{
+			ServiceName:  "thanos",
+			SamplerType:  "const",
+			SamplerParam: 1,
+			Endpoint:     "http://" + newJaeger.InternalEndpoint("jaeger.thrift") + "/api/traces", // which endpoint to use here?
+		},
+	})
+	testutil.Ok(t, err)
+
+	prom1, sidecar1 := e2ethanos.NewPrometheusWithJaegerTracingSidecarCustomImage(env, "alone", e2ethanos.DefaultPromConfig("prom-alone", 0, "", "", e2ethanos.LocalPrometheusTarget), "",
+		e2ethanos.DefaultPrometheusImage(), "", e2ethanos.DefaultImage(), string(jaegerConfig), "")
+	testutil.Ok(t, e2e.StartAndWaitReady(prom1, sidecar1))
+
+	qb := e2ethanos.NewQuerierBuilder(env, "1", sidecar1.InternalEndpoint("grpc"))
+	q := qb.WithTracingConfig(fmt.Sprintf(`type: JAEGER
+config:
+  sampler_type: const
+  sampler_param: 1
+  service_name: %s
+  endpoint: %s`, qb.Name(), "http://"+newJaeger.InternalEndpoint("jaeger.thrift")+"/api/traces")).Init()
+	testutil.Ok(t, e2e.StartAndWaitReady(q))
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	t.Cleanup(cancel)
+
+	queryAndAssertSeries(t, ctx, q.Endpoint("http"), e2ethanos.QueryUpWithoutInstance, time.Now, promclient.QueryOptions{
+		Deduplicate: false,
+	}, []model.Metric{
+		{
+			"job":        "myself",
+			"prometheus": "prom-alone",
+			"replica":    "0",
+		},
+	})
+
+	url := "http://" + strings.TrimSpace(newJaeger.Endpoint("http")+"/api/traces?service=thanos") // pretty print
+	// fmt.Printf("\n\n JAEGER URL: %s", url) // --> gives the correct URL displaying 'thanos' traces.
+	request, err := http.NewRequest("GET", url, nil)
+	testutil.Ok(t, err)
+	client := &http.Client{}
+
+	ctx = context.Background()
+	b := backoff.New(ctx, backoff.Config{
+		Min:        500 * time.Millisecond,
+		Max:        5 * time.Second,
+		MaxRetries: 10,
+	})
+	for b.Reset(); b.Ongoing(); {
+		response, err := client.Do(request)
+		// Retry if we have a connection problem (timeout, etc)
+		if err != nil {
+			b.Wait()
+			continue
+		}
+
+		// Jaeger might give a 404 or 500 before the trace is there.  Retry.
+		if response.StatusCode != http.StatusOK {
+			t.Logf("\n\n WAITING")
+			b.Wait()
+			continue
+		}
+
+		// We got a 200 response.
+		defer response.Body.Close()
+
+		body, err := ioutil.ReadAll(response.Body)
```

Comment:
We could potentially also assert on if response has expected content (e.g. is trace ID present)

Reply:
That's where the issue comes in.
For some reason, I keep getting empty response when I run this test locally.

Reply:
I'm seeing some panic in the CI logs, it's probably related:
```
12:30:43 querier-1: panic: runtime error: invalid memory address or nil pointer dereference
12:30:43 querier-1: [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1647a96]
12:30:43 querier-1: goroutine 1 [running]:
12:30:43 querier-1: go.opentelemetry.io/otel/exporters/jaeger.New({0x0, 0x0})
12:30:43 querier-1: /home/runner/go/pkg/mod/go.opentelemetry.io/otel/exporters/jaeger@v1.7.0/jaeger.go:47 +0x36
12:30:43 querier-1: github.com/thanos-io/thanos/pkg/tracing/jaeger.NewTracerProvider({0x21ea6c0, 0xc000054018}, {0x21ac500, 0xc000098190}, {0xc0000c9000, 0x3d, 0x40})
12:30:43 querier-1: /home/runner/work/thanos/thanos/pkg/tracing/jaeger/jaeger.go:62 +0x125
12:30:43 querier-1: github.com/thanos-io/thanos/pkg/tracing/client.NewTracer({0x21ea6c0, 0xc000054018}, {0x21ac500, 0xc000098190}, 0x7ffc7fbedd6f, {0xc0000da9c0, 0x57, 0x60})
12:30:43 querier-1: /home/runner/work/thanos/thanos/pkg/tracing/client/factory.go:68 +0x2e6
12:30:43 querier-1: main.main()
12:30:43 querier-1: /home/runner/work/thanos/thanos/cmd/thanos/main.go:102 +0xd71
```

Reply:
Thanks!
The CI test passes now but my E2E tests are failing.